master/internal/rm/agentrm/provisioner/gcp/gcp.go

```diff
@@ -164,7 +167,14 @@ func (c *gcpCluster) stateFromInstance(inst *compute.Instance) model.InstanceSta
 }
 
 func (c *gcpCluster) generateInstanceNamePattern() string {
-	return c.config.NamePrefix + petname.Generate(2, "-") + "-#####"
+	genName := c.config.NamePrefix + petname.Generate(2, "-")
+	suffix := "-#####"
+	// We make sure that the generated name is less than the max length
+	if len(genName) > maxInstanceNameLength-len(suffix) {
+		c.syslog.Warnf("Generated name for instance was truncated")
```

Comment:
```suggestion
		c.syslog.WithField("genName", genName).Warnf("Generated name for instance was truncated")
```
nit: might be good to add some more info about the node name